### PR TITLE
Fix crash when a key containing periods doesn't exist. Fixed shared preferences null exception during unit tests

### PR DESCRIPTION
**Pull request by acoutts**

### CODE REVIEW

File: lib/src/translations.dart [hunk 1]
```diff
@@ -20,11 +32,12 @@ class Translations {
       if (value is Map<String, dynamic>) value = value[keys[i]];
     }
 
-    if (value == null) {
-      throw Exception('Cannot cache a key that is not nested.');
```
Comment (acoutts) [inline lib/src/translations.dart:24]: The problem with this throw is it's an uncaught exception so it crashes the entire app.

File: test/easy_localization_context_test.dart [hunk 2]
```diff
@@ -38,8 +38,8 @@ class MyWidget extends StatelessWidget {
 }
 
 void main() async {
-  await EasyLocalization.ensureInitialized();
   SharedPreferences.setMockInitialValues({});
+  await EasyLocalization.ensureInitialized();
```
Comment (acoutts) [inline test/easy_localization_context_test.dart:42]: This fixes the test failure due to the shared preferences null exception. Have to initialize shared preferences mock first.

File: lib/src/translations.dart (hunk 1 above)
Comment (aissat) [reply lib/src/translations.dart:24]: yeah agreed with u  :100:

**Review by aissat — APPROVED** [review]
@acoutts :100:

**Comment by acoutts** [comment]
LGTM! 😄

**Review by Overman775 — APPROVED** [review]

**Comment by Overman775** [comment]
@acoutts  thanks for PR, merget 🚀

**Pull request merged**